src/components/capture/GenerateButton.tsx

```diff
@@ -32,14 +36,29 @@ function CaptureGenerateButton({
         { initiateDiscovery: createWorkflowMetadata?.initiateDiscovery }
     );
 
+    // Details Form Store
+    const selectedConnectorId = useDetailsForm_connectorImage_connectorId();
+    const previousConnectorId =
+        useDetailsForm_previousConnectorImage_connectorId();
+
+    // Form State Store
     const formStatus = useFormStateStore_status();
+
     useEffect(() => {
         if (createWorkflowMetadata?.initiateDiscovery) {
-            if (formStatus === FormStatus.GENERATED) {
+            if (
```

Comment:
In retrospect, this `if` statement nesting is syntactically superfluous. The condition of the outer `if` statement can be extended to encapsulate the condition of the inner `if` statement.